Refactor Kiali documentation (#8461)
kyma-project/kyma: docs/kiali/05-01-kiali.md

| Parameter | Description | Default value |
|-----------|-------------|---------------|
| **server.webRoot** | Defines the context root path for Kiali console, API endpoints, and readiness probes. | `/` |
| **deployment.viewOnlyMode** | When set to `true`, Kiali is available in view-only mode, allowing you to view and retrieve management data for the Service Mesh. You cannot modify the Service Mesh.  | `true` |
| **deployment.accessibleNamespaces** | Specifies the Namespaces Kiali can access to monitor the Service Mesh components deployed there. You can provide the names using regex expressions. The default value is `**`(two asterisks) meaning Kiali can access any Namespace. | `**` |

> | **deployment.resources.requests.cpu** | Defines the minimum CPU that the kiali operator will use. | `10m` |

Reviewer: ```suggestion
| **deployment.resources.requests.cpu** | Minimum number of CPUs requested by the Kiali operator to use. | `10m` |
```

> | **deployment.resources.requests.memory** | Defines the minimum amount of memory that the kiali operator will use. | `20Mi` |

Reviewer: ```suggestion
| **deployment.resources.requests.memory** | Minimum amount of memory requested by the Kiali operator to use. | `20Mi` |
```

> | **deployment.resources.limits.cpu** | Defines the maximum CPU that can be used by the kiali operator. | `100m` |

Reviewer: ```suggestion
| **deployment.resources.limits.cpu** | Maximum number of CPUs available for the Kiali operator to use. | `100m` |
```

> | **deployment.resources.limits.memory** | Defines the maximum amount of memory that can be used by the kiali operator. | `100Mi` |

Reviewer: ```suggestion
| **deployment.resources.limits.memory** | Maximum amount of memory available for the Kiali operator to use. | `100Mi` |
```

> | **deployment.kubernetes_config.qps** | Defines the allowed queries per second to adjust the API server throttling rate. | `50` |

Reviewer: ```suggestion
| **deployment.kubernetes_config.qps** | Defines the allowed queries per second to adjust the API server's throttling rate. | `50` |
```

| **deployment.accessibleNamespaces** | Specifies the Namespaces Kiali can access to monitor the Service Mesh components deployed there. You can provide the names using regex expressions. The default value is `**`(two asterisks) meaning Kiali can access any Namespace. | `**` |

> 

Reviewer 2: In regards to the config options above:
- metrics.enabled - I prefer to not mention as we have it disabled by default (I think it was not working proper)
- server.webroot - would you ever change it?

Instead check the other attributes of the kiali CRD which might be relevant like:
- resources - to increase memory limit for exampe
- kubernetes_config.qps - to adjust apiserver throttling rate

Author: I will check to add the extra configurations that might be useful.

On those to remove:
- `metrics.enabled`: will remove it.
- `server.webroot`: might be useful to use a dedicated path. In any case it is an option that exists and does not harm to document it I think.

Author: Removed the `metrics.enabled` and added the suggested config values.

> For details on Kiali configuration and customization, see [this](https://www.kiali.io/documentation/getting-started/) documentation.

Reviewer 2: I think it is more important to link the CRD documentation at https://github.com/kiali/kiali-operator/blob/master/deploy/kiali/kiali_cr.yaml or simply the values.yaml of kyma https://github.com/kyma-project/kyma/blob/master/resources/kiali/values.yaml

Author: Changed this line, now links to both the CRD and the values files on the kyma repository.
